src/transform/src/use_indexes.rs

```diff
@@ -41,19 +37,35 @@ impl crate::Transform for FilterEqualLiteral {
         relation: &mut RelationExpr,
         args: TransformArgs,
     ) -> Result<(), crate::TransformError> {
-        self.transform(relation, args);
+        let mut indexes = HashMap::new();
+        for (on_id, idxs) in args.indexes {
+            let keys = idxs.iter().map(|(_id, keys)| keys.clone()).collect();
+            indexes.insert(*on_id, keys);
+        }
+        self.transform(relation, &indexes);
         Ok(())
     }
 }
 
 impl FilterEqualLiteral {
-    pub fn transform(&self, relation: &mut RelationExpr, args: TransformArgs) {
+    /// Replaces filters of the form ScalarExpr::Column(i) == ScalarExpr::Literal, where i is a column for
+    /// which an index exists, with a
+    /// Join{
+    ///   equivalences: [(0, i), (1,0)],
+    ///   ArrangeBy{input, keys: [ScalarExpr::Column(i)]},
+    ///   <constant>
+    /// }
```

Comment:
I think, looking at the transform, that the comments here are now out of date. No worries, and can fix once we are happy with the code in its final state, but wanted to flag! :D